src/Main.elm

```diff
@@ -0,0 +1,71 @@
+module Main exposing (..)
+
+import Api exposing (..)
+import Browser
+import Dict
+import Html.Styled exposing (toUnstyled)
+import Model exposing (..)
+import RemoteData exposing (RemoteData(..), fromResult)
+import View exposing (viewStyled)
+
+
+main : Program () Model Msg
+main =
+    Browser.element
+        { init = always ( initModel, Cmd.batch [ fetchUsers ] )
+        , update = update
+        , subscriptions = always Sub.none
+        , view = viewStyled >> toUnstyled
+        }
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        GetUsersResponse result ->
+            ( { model | users = fromResult result }, Cmd.none )
+
+        UpsertUserResponse result ->
+            case ( result, model.users ) of
+                ( Ok ( id, user ), Success theUsers ) ->
+                    ( { model | users = Success (theUsers |> Dict.insert id user), underEdit = NotEditing }, Cmd.none )
+
+                ( Err errs, _ ) ->
+                    ( { model | errors = errs :: model.errors }, Cmd.none )
```

Comment:
Might be nicer with specifying the action that resulted in this error. In one of our apps we've made we kind of went bonkers with it: reproducing the request as well as the response in the "hey something went wrong" dialog.

<img width="605" alt="Screenshot 2022-01-21 at 18 07 44" src="https://user-images.githubusercontent.com/149425/150569795-df9c5d90-6214-4d13-9cfb-5722e446a56f.png">
 
<img width="696" alt="Screenshot 2022-01-21 at 18 07 49" src="https://user-images.githubusercontent.com/149425/150569803-3226666b-cf6e-49b8-a3b4-550b6bdd86d6.png">

But anyways this wasn't a part of the requirements for this challenge - your solutions works :) and by not giving the user a chance to fire another request before the error from previous one gets dismissed, it's mostly fine. Although they still can eg. focus on another user medaillon and the error doesn't say anything to identify which user it was for.

Reply:
Good point about refocusing